metadata-ingestion/tests/unit/test_tableau_source.py

```diff
@@ -247,3 +248,23 @@ def test_optimize_query_filter_handles_no_duplicates():
     assert len(result) == 2
     assert result[c.ID_WITH_IN] == ["id1", "id2"]
     assert result[c.PROJECT_NAME_WITH_IN] == ["project1", "project2"]
+
+
+def testTableaUpstreamReference():
```

Comment:
following snake case strategy...

```suggestion
def test_tableau_upstream_reference():
```